### ISSUE TITLE
Feature Request : Capability to render GitHub flavor Markdown comments

### ISSUE THREAD

**Issue by AdarshRawat1** [doxygen/doxygen#10959]
## Describe the Feature
 
Currently, if I want to pass any Doxygen command in Markdown format, it is rendered in GitHub, thus making the output a bit unpleasant.

## Example

Suppose I want to use page command, I will have to use `\page` command or PHP Markdown Extra, which will also be visible in GitHub. 
 
## Possible Solution 
Add a way to pass commands in Markdown files without actually messing up the output of files in GitHub / GitLab flavor markdown. This can make the whole documentation cleaner.
## Additional context

GitHub supports following comments 
- HTML comments  ``` <!-- .... --> ```
  -  This is not rendered by Doxygen 
  

- `[//]: <> (This is command )` , `[//]: # (This is also a command in comment )` , `[//]: # "Comment"`
  - Doxygen processes these comments, the command works only if they are on the next line. Additionally, after processing the comment, there is a `)` in the output when using `[//]: # ()` . 
  
  
  
## To Reproduce

Use any Doxygen command in Markdown format and see it being rendered on GitHub.

## Expected behavior
The command should not be visible in GitHub 

## Version
Applies to all version [I tried 1.11.0 also ]

**Comment by albert-github** [comment]
So what you would like to have is something like
```
[//]: <> (\page pg1 the first page)
This is the page body
```
which in GitHub would render just like the line:
```
This is the page body
```

and in doxygen like
```
\page pg1 the first page
This is the page body
```

Be aware that in this case not all doxygen commands can be honored (due the order things are processed)-

(Did you also file an enhancement request with GitHub and GitLab?)

**Comment by albert-github** [comment]
@AdarshRawat1 

Is there anywhere a formal specification of this feature as in the https://github.github.com/gfm/  I don't see any reference to `[//]`

**Comment by AdarshRawat1** [comment]
> So what you would like to have is something like
> 
> ```
> [//]: <> (\page pg1 the first page)
> This is the page body
> ```
> 
> which in GitHub would render just like the line:
> 
> ```
> This is the page body
> ```
> 
> and in doxygen like
> 
> ```
> \page pg1 the first page
> This is the page body
> ```

Yes ! 

> Be aware that in this case not all doxygen commands can be honored (due the order things are processed)-

So, you mean that command combination like [`\htmlonly`](https://www.doxygen.nl/manual/commands.html#cmdhtmlonly) and [`\endhtmlonly`](https://www.doxygen.nl/manual/commands.html#cmdendhtmlonly) won't work ??

> (Did you also file an enhancement request with GitHub and GitLab?)

Not yet, but I will do it soon. That's a good idea!

**Comment by AdarshRawat1** [comment]
> @AdarshRawat1
> 
> Is there anywhere a formal specification of this feature as in the https://github.github.com/gfm/ I don't see any reference to `[//]`

No, I couldn't find it, I tried various comment methods and these are the only ones supported by GitHub flavor.

It does work tho, I was able to Make TOC with it. But it leads to a `)` in [output ](https://adarshrawat1.github.io/P4c-Documentation-Hosting/md__2home_2runner_2work_2_p4c-_documentation-_hosting_2_p4c-_documentation-_hosting_2backends_2dpdk_2_r_e_a_d_m_e.html).

https://github.com/AdarshRawat1/P4c-Documentation-Hosting/blob/583d6c071d4e4056cd8eefef20c7a0a39f842b5c/backends/dpdk/README.md?plain=1#L1-L5

**Comment by albert-github** [comment]
> So, you mean that command combination like [\htmlonly](https://www.doxygen.nl/manual/commands.html#cmdhtmlonly) and [\endhtmlonly](https://www.doxygen.nl/manual/commands.html#cmdendhtmlonly) won't work ??

I'm not sure which commands won't work and it depends where the feature will be implemented (at the moment I'm thinking about some handling in commentcnv but this is very preliminary thought). 
Thanks for hinting to `\htmlonly` / ´\endhtmlonly` something to think about as well (as well as about other "block" commands).

**Comment by AdarshRawat1** [comment]
> It does work tho, I was able to Make TOC with it. But it leads to a `)` in [output ](https://adarshrawat1.github.io/P4c-Documentation-Hosting/md__2home_2runner_2work_2_p4c-_documentation-_hosting_2_p4c-_documentation-_hosting_2backends_2dpdk_2_r_e_a_d_m_e.html).
> 
> https://github.com/AdarshRawat1/P4c-Documentation-Hosting/blob/583d6c071d4e4056cd8eefef20c7a0a39f842b5c/backends/dpdk/README.md?plain=1#L1-L5



[`\pages`](https://github.com/AdarshRawat1/P4c-Documentation-Hosting/blob/52b68e623331e0325e17e88f1107920a3ee16b28/docs/README.md?plain=1#L1-L3
)  also works but leads to similar [output 
](https://adarshrawat1.github.io/P4c-Documentation-Hosting/_sample.html)

**Comment by AdarshRawat1** [comment]
> (Did you also file an enhancement request with GitHub and GitLab?)

Any suggestions on what to include ? is there any comment style for Markdown that Doxygen renders?

**Comment by albert-github** [comment]
Not really a suggestion but they could implement the commands as specified in the doxygen manual, especially in https://doxygen.nl/manual/commands.html

**Comment by doxygen** [comment]
Maybe we could add something like `<!--! This interpreted by doxygen but hidden by HTML/Markdown viewers -->`
where doxygen interprets the second `!` as an instruction to 'undo' the hiding.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10966

**Comment by AdarshRawat1** [comment]
@albert-github , @doxygen Thank you for the prompt response.

This will be very helpful for projects having high dependency on GitHub markdown for documentation.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by AdarshRawat1** [comment]
Thanks @albert-github

**Comment by AdarshRawat1** [comment]
@albert-github @doxygen When can we expect a subversion release with this feature?

**Comment by MarkCallow** [comment]
@AdarshRawat1 have you thought about creating a GitHub Action that runs Doxygen to generate HTML from your documentation and deploys the result to your GitHub Pages?

**Comment by AdarshRawat1** [comment]
> @AdarshRawat1 have you thought about creating a GitHub Action that runs Doxygen to generate HTML from your documentation and deploys the result to your GitHub Pages?

Yes !

> EDIT: It seems to me that you can cause GitHub to produce a very misleading rendering of your documentation with this feature depending on the commands you are using. For example, if you see the command `\page` before it you can understand the meaning of "authors Authors", to pick something from one of my own Markdown documentation pages. That appearing on a line by itself is just weird and looks like a typo. I'm not sure supporting this feature is a good idea.

@MarkCallow Thanks for brining this up. I'm new to this, I'm not sure if it will have negative impacts.
Is there a better way to achieve this ?

**Comment by iamAyushChamoli** [comment]
I'm also waiting for this @albert-github @doxygen

**Comment by MarkCallow** [comment]
> @MarkCallow Thanks for brining this up. I'm still exploring this idea, I'm not sure if it will have negative impacts.
> Is there a better way to achieve this ?

How about using a different file extension for Doxygen markdown files such as `.dmd`. Then GitHub will not apply its markdown renderer. 

> Add a way to pass commands in Markdown files without actually messing up the output of files in GitHub / GitLab flavor markdown. This can make the whole documentation cleaner.

But it will likely be "messed up" anyway with the *effects* of the Doxygen commands missing. I suspect people who use this will spend (waste?) a lot of time trying to make their documents look good and understandable in both GitHub's rendering, minus the Doxygen commands, and in Doxygen's rendering. To take another instance from my own documentation

```
@snippet{doc} version.h API version
```

Rendered without the command it will be the meaningless and question-raising

```
version.h API version
```

So you might decide to hide the entire line in the special comment. I refer you back to my time-wasting point.

Personally I find the presence of the commands helpful to understanding what this "markdown" document really is.

Fortunately GitHub does not apply its markdown rendering when showing diffs of `.md` files.

**Comment by MarkCallow** [comment]
> But this will remove the rendered file from GitHub. I want to maintain the repository as well as the generated Docs.

I do not understand what you mean. Changing the extension will not remove the source file from the repository. The Doxygen rendered file should not be saved in the repository. What changing the extension will do is make GitHub show the document the same way it would show any source file.

> This feature will not make it compulsory to give commands in comments, such as  . It will just process the commands in comments as well. Which shouldn't be an issue as you have the freedom to use the comments or not.

I understand that. I am giving my opinion as to some of the issues that are likely to arise when this feature is used. I will not be using it. I spend long enough already getting the Doxygen output to look the way I want.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
